File: ljh/level2/멀쩡한 사각형/Solution.js
function solution(w, h) {
    if (w === h) {
        return w * h - w;
    }
    
    let count = gcd(w, h); // 단위의 개수
    let unit = 0;  // 한 단위에서 대각선이 지나는 칸의 개수
    let unit_w = w / count; // 한 단위의 w 길이
    while (unit_w > 0) {
        unit += Math.ceil(unit_w * h / w) - Math.floor((unit_w - 1) * h / w);
        unit_w--;
    }

    return w * h - unit * count;
}

function gcd(w, h) {
    if (h === 0) {
        return w;
    } else {
        return gcd (h, w%h);
    }
}

File: ljh/level2/멀쩡한 사각형/Solution.java
class Solution {
    public long solution(int w, int h) {
        if (w == h) 
            return (long)w * h - w;
        
        int temp;
        if (h > w) {
            temp = w;
            w = h;
            h = temp;
        }
        
        int gcd = gcd(w,h);

        double unit = 0;
        double unitH = (double) h / gcd;
        for (double k = 1; k <= unitH; k++) {
            unit += (Math.ceil(k * w / h) - Math.floor((k - 1) * w / h));
        }

        return (long) w * h - (long) unit * gcd;
    }
    
    public int gcd(int w, int h) {
        int r;
        
        while (h > 0) {
            r = w % h;
            w = h;
            h = r;
        }
        
        return w;    
    }
}